tests/features/federation.feature

```diff
@@ -0,0 +1,158 @@
+    And Collabora fetches checkFileInfo
+    And checkFileInfo "BaseFileName" is "file-reshare-readonly.odt"
+    And checkFileInfo "UserCanWrite" is false
+    And both Collabora files used the same file id
+    And Collabora can not save the file with the content of "./../assets/template.odt"
+    Then Collabora downoads the file and it is equal to "./../assets/template.ods"
+
+  Scenario: Share a file by federation and reshare it with write permissions
+    Given on instance "serverA"
+    And as user "user1"
+    And User "user1" uploads file "./../assets/template.odt" to "/file-reshare-rw.odt"
+    And share the file "/file-reshare-rw.odt" as a federated share to "user2" on "serverB"
+
+    Given on instance "serverB"
+    And as user "user2"
+    And user "user2" accepts last share
+    Then User "user2" opens "/file-reshare-rw.odt"
+    And Collabora fetches checkFileInfo
+    And checkFileInfo "BaseFileName" is "file-reshare-rw.odt"
+    And checkFileInfo "UserCanWrite" is true
+    Then Collabora downoads the file and it is equal to "./../assets/template.odt"
+    And Collabora can save the file with the content of "./../assets/template.ods"
+
+    And as "user2" create a share with
+      | path | /file-reshare-rw.odt |
+      | shareType | 0 |
+      | shareWith | user3 |
+      | permissions | 31 |
+
+    Given as user "user3"
+    When User "user3" opens "/file-reshare-rw.odt"
+    And Collabora fetches checkFileInfo
+    Then checkFileInfo "BaseFileName" is "file-reshare-rw.odt"
+    And checkFileInfo "UserCanWrite" is true
+    And both Collabora files used the same file id
+    And Collabora can save the file with the content of "./../assets/template.ods"
+    And Collabora downoads the file and it is equal to "./../assets/template.ods"
+
+  Scenario: Share a file by federation and reshare it as a link
+    Given on instance "serverA"
+    And as user "user1"
+    And User "user1" uploads file "./../assets/template.odt" to "/file-reshare-link.odt"
+    And share the file "/file-reshare-link.odt" as a federated share to "user2" on "serverB"
+    Then User "user1" opens "/file-reshare-link.odt"
+    And Collabora fetches checkFileInfo
+
+
+    Given on instance "serverB"
+    And as user "user2"
+    And user "user2" accepts last share
+    Then User "user2" opens "/file-reshare-link.odt"
+    And Collabora fetches checkFileInfo
+    And Collabora receives the following in the checkFileInfo response
+      | BaseFileName | file-reshare-link.odt |
+      | UserCanWrite | 1                     |
+    And checkFileInfo "BaseFileName" is "file-reshare-link.odt"
+    And checkFileInfo "UserCanWrite" is true
+    And both Collabora files used the same file id
+    Then Collabora downoads the file and it is equal to "./../assets/template.odt"
+    And Collabora can save the file with the content of "./../assets/template.ods"
+
+#    # This might fail curently due to using the same instance
+#    And as "user2" create a share with
```

Comment:
This part seems to have an issue due to reusing the same instance for the federated share, I'll investigate further in a follow up and then also make sure to bring up two real isntances in the github action.